services/cloudfront.py

```diff
@@ -2,40 +2,137 @@
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_no_deprecated_ssl_protocols():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        for origin in distribution["DistributionConfig"]["Origins"]["Items"]:
+            if (
+                "CustomOriginConfig" in origin
+                and "SSLv3" in origin["CustomOriginConfig"]["OriginSslProtocols"]["Items"]
+            ):
+
+                non_compliant_resources.append(distribution["ARN"])
+                break
+        else:
+            compliant_resources.append(distribution["ARN"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_s3_origin_access_control_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]
+
+    for distribution in distributions["Items"]:
+        for origin in distribution["Origins"]["Items"]:
+            if "S3OriginConfig" in origin and origin["OriginAccessControlId"] == "":
+                non_compliant_resources.append(distribution["Id"])
+                break
+        else:
+            compliant_resources.append(distribution["Id"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def cloudfront_viewer_policy_https():
+    compliant_resources = []
+    non_compliant_resources = []
+    distributions = client.list_distributions()["DistributionList"]["Items"]
+
+    for distribution in distributions:
+        distribution = client.get_distribution(Id=distribution["Id"])["Distribution"]
+
+        if distribution["DistributionConfig"]["DefaultCacheBehavior"]["ViewerProtocolPolicy"] != "allow-all":
+            for behavior in distribution["DistributionConfig"]["CacheBehaviors"]["Items"]:
+                if behavior["ViewerProtocolPolicy"] == "allow-all":
+                    non_compliant_resources.append(distribution["ARN"])
+                    break
+            else:
+                compliant_resources.append(distribution["ARN"])
+        else:
+            non_compliant_resources.append(distribution["ARN"])
```

Comment:
flow control을 조금 더 단순하게 작성할 수 있습니다.
```suggestion
    for distribution in distributions:
        if distribution["DefaultCacheBehavior"]["ViewerProtocolPolicy"] == "allow-all":
            non_compliant_resources.append(distribution["ARN"])
            continue

        allow_alls = [
            behavior
            for behavior in distribution["CacheBehaviors"]["Items"]
            if behavior["ViewerProtocolPolicy"] == "allow-all"
        ]
        if allow_alls:
            non_compliant_resources.append(distribution["ARN"])
            continue

        compliant_resources.append(distribution["ARN"])
```